### ISSUE TITLE
API Review

### ISSUE THREAD

**Issue by bitwalker** [bitwalker/timex#9]
Timex has undergone a great deal of change over the past few months. Before we can start thinking about the possibility of merging it into the standard library, I think it's important to take a step back and think about the API, and make sure it meets the following requirements:

- [x] User guide/intro documentation
- [x] Fully documented (`@moduledoc`, `@doc`, `@spec`)
- [x] Fully tested (a good suite of tests for each API function)
- [x] Consistency (is there uniformity to the order of arguments, can pipelines be used when chaining date/time manipulations together, is naming consistent, are modules namespaced consistently, etc).
- [ ] Dialyzer gods are pleased.

These are all "ideal" items to be checked off before a potential merge into the standard library can be considered. These aren't strictly tasks, so much as they are a confirmation that these items have been reviewed, and are deemed satisfactory.

**Comment by mr-bt** [comment]
@bitwalker are you thinking in moving to R18 new time API?

**Comment by bitwalker** [comment]
@mr-bt I've actually already implemented this in the `Time` module. If you look at the definitions for `now/1` or `measure/3` you can see where I'm checking a cached value for the current OTP release, and using the new Time API if it's available (i.e. OTP 18+). The `Date` module does not use the Time API (it really only deals with the Erlang `calendar` module.

I'm thinking of refactoring the way I'm handling whether to use the new Time API or not, because I really don't like having to check a cached value, but it was the only way I could think of to handle the case where you might compile on a different version than you run in prod. I'll probably change it to bake in one or the other at compile time, and if you deploy to an older version of OTP, then it will explode, but that should probably be expected anyway.
